test/integration/http2_integration_test.cc

```diff
@@ -195,6 +195,110 @@ TEST_P(Http2IntegrationTest, Trailers) { testTrailers(1024, 2048); }
 
 TEST_P(Http2IntegrationTest, TrailersGiantBody) { testTrailers(1024 * 1024, 1024 * 1024); }
 
+TEST_P(Http2IntegrationTest, IdleTimoutBasic) { testIdleTimeoutBasic(); }
+
+TEST_P(Http2IntegrationTest, IdleTimeoutWithTwoRequests) { testIdleTimeoutWithTwoRequests(); }
+
+TEST_P(Http2IntegrationTest, IdleTimeoutWithSimultaneousRequests) {
+  simultaneousRequestsWithIdleTimeout();
+}
+
+// Interleave two requests and responses and make sure that idle timeout is handled correctly.
+void Http2IntegrationTest::simultaneousRequestsWithIdleTimeout() {
+  FakeHttpConnectionPtr fake_upstream_connection1;
+  FakeHttpConnectionPtr fake_upstream_connection2;
+  Http::StreamEncoder* encoder1;
+  Http::StreamEncoder* encoder2;
+  IntegrationStreamDecoderPtr response1(new IntegrationStreamDecoder(*dispatcher_));
+  IntegrationStreamDecoderPtr response2(new IntegrationStreamDecoder(*dispatcher_));
+  FakeStreamPtr upstream_request1;
+  FakeStreamPtr upstream_request2;
+  int32_t request1_bytes = 1024;
+  int32_t request2_bytes = 512;
+
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+    auto* static_resources = bootstrap.mutable_static_resources();
+    auto* cluster = static_resources->mutable_clusters(0);
+    auto* http_protocol_options = cluster->mutable_common_http_protocol_options();
+    auto* idle_time_out = http_protocol_options->mutable_idle_timeout();
+    std::chrono::milliseconds timeout(1000);
+    auto seconds = std::chrono::duration_cast<std::chrono::seconds>(timeout);
+    idle_time_out->set_seconds(seconds.count());
+  });
+
+  initialize();
+
+  // Validate that timeout is set and is as per expectation.
+  bool has_idle_timeout =
+      config_helper_.bootstrap().static_resources().clusters(0).has_common_http_protocol_options();
+  EXPECT_TRUE(has_idle_timeout);
```

Comment:
No need to validate this; this is just validating that previous test code in this function worked.